### PR TITLE
BaSP-32: Super-Admins shared components implementation

### PR DESCRIPTION
**Pull request by Martinzancocchia**
> > ### **IMPLEMENTING SHARED COMPONENTS IN SUPER ADMIN**
> > 
> > **[DONE] _Table_**
> > 
> >
> Screenshot 2022-11-10 230959
> 
> > 
> > **[DONE] _Modal_**
> > 
> > 
> > 
> > 
> > 
> > 
> > **[DONE] _Form_**
> > 
> > 
> > 
> > 
> > **[DONE] _Imput_**
> > 
> > 
> > 
> > 
> > 
> > 
>

### CODE REVIEW

**Comment by Baldizan** [comment]
The arrow images were also added on TG-31. It would be better to remove them, otherwise files may duplicate.

File: src/Components/Shared/Table/Row/row.module.css [hunk 1]
```diff
@@ -0,0 +1,14 @@
+ .tr {
```
[the remark this answers is not in the document]
Comment (Martinzancocchia) [reply src/Components/Shared/Table/Row/row.module.css:1]: done!

**Comment by Martinzancocchia** [comment]
> The arrow images were also added on TG-31. It would be better to remove them, otherwise files may duplicate.
Done!!!

File: src/Components/SuperAdmins/Form/index.js [hunk 2]
```diff
@@ -1,179 +1,143 @@
...
     lastName: '',
     email: '',
     password: ''
   });
+  const [modalDisplay, setModalDisplay] = useState(false);
+  const [modalContent, setModalContent] = useState({ message: '', error: '' });
+  setModalContent;
+
+  useEffect(() => {
+    console.log(selectedSuperAdmin);
```
Comment (Baldizan) [inline src/Components/SuperAdmins/Form/index.js:23]: please remove this console log

**Review by Baldizan — APPROVED** [review]

File: src/Components/Shared/Table/Row/row.module.css [hunk 3]
```diff
@@ -1,15 +1,14 @@
 .tr {
-  display: flex;
-  width: 100%;
+  min-width: 150px;
 }
 
 .td {
-  width: 100%;
+  min-width: 150px;
```
Comment (LucianaPrat) [inline src/Components/Shared/Table/Row/row.module.css:6]: There should be no changes to this shared component

File: src/Components/Shared/Table/index.js [hunk 4]
```diff
@@ -20,14 +20,13 @@ const Table = ({ data, headers, editItem, deleteItem }) => {
               key={item._id}
               rowItem={item}
               headers={headers}
-              editItem={editItem}
-              deleteItem={deleteItem}
+              editItem={() => editItem(item)}
```
Comment (LucianaPrat) [inline src/Components/Shared/Table/index.js:23]: There should be no changes to this shared component

File: src/Components/Shared/Table/table.module.css [hunk 5]
```diff
@@ -1,7 +1,21 @@
-.tr {
+.container {
   display: flex;
-  padding: 10px 10px;
-  border: solid 0.5px #eeeeee;
+  flex-direction: column;
+  justify-content: flex-start;
```
Comment (LucianaPrat) [inline src/Components/Shared/Table/table.module.css:4]: There should be no changes to this shared component

File: src/Components/SuperAdmins/Form/index.js [hunk 6]
```diff
@@ -1,179 +1,142 @@
...
-        message={message}
-        titleModal={titleModal}
-      />
+      <Form onSubmit={onSubmit} title={titleForm}>
+        <Input
+          onChange={onChange}
+          placeholder={'Enter your name'}
+          value={superAdminInput.name}
+          name="name"
+          title={'Name'}
```
Comment (LucianaPrat) [inline src/Components/SuperAdmins/Form/index.js:103]: should be Title = "Name" , to be consistent with the line above

File: src/Components/SuperAdmins/Form/index.js [hunk 7]
```diff
@@ -1,179 +1,142 @@
...
+          required
+        />
+        <Input
+          onChange={onChange}
+          placeholder={'Enter your last name'}
+          value={superAdminInput.lastName}
+          name="lastName"
+          title={'Last Name'}
```
Comment (LucianaPrat) [inline src/Components/SuperAdmins/Form/index.js:111]: should be Title = "Name" , to be consistent with the line above

File: src/Components/SuperAdmins/Form/index.js [hunk 8]
```diff
@@ -1,179 +1,142 @@
...
+          title={'Email'}
+          required
+        />
+        <Input
+          onChange={onChange}
+          placeholder={'Enter a password'}
+          value={superAdminInput.password}
+          type="password"
+          name="password"
+          title={'Password'}
```
Comment (LucianaPrat) [inline src/Components/SuperAdmins/Form/index.js:128]: should be Title = "Name" , to be consistent with the line above

File: src/Components/SuperAdmins/Form/index.js [hunk 9]
```diff
@@ -1,179 +1,142 @@
...
+          name="lastName"
+          title={'Last Name'}
+          required
+        />
+        <Input
+          onChange={onChange}
+          placeholder={'Enter a valid email address'}
+          value={superAdminInput.email}
+          name="email"
+          title={'Email'}
```
Comment (LucianaPrat) [inline src/Components/SuperAdmins/Form/index.js:119]: should be Title = "Name" , to be consistent with the line above

File: src/Components/SuperAdmins/super-admins.module.css [hunk 10]
```diff
@@ -1,10 +1,9 @@
...
-  flex: 1;
+.container h2 {
+  text-align: center;
+
+}.container {
   display: flex;
-  justify-content: center;
   flex-direction: column;
+  justify-content: space-evenly;
   align-items: center;
```
Comment (LucianaPrat) [inline src/Components/SuperAdmins/super-admins.module.css:8]: .container should go one line below , align-item should go below flex-direction

File: src/Components/SuperAdmins/index.js [hunk 11]
```diff
@@ -1,51 +1,110 @@
...
+        />
+      </div>
+      {successModalDisplay ? (
+        <Modal
+          heading={`${selectedItem.name} ${selectedItem.lastName} deleted successfully!`}
+          setModalDisplay={setSuccessModalDisplay}
+          theme={'success'}
+        />
+      ) : null}
+      {modalDisplay && (
```
Comment (LucianaPrat) [inline src/Components/SuperAdmins/index.js:92]: should be {modalDisplay && (.....)}

**Review by LucianaPrat — CHANGES_REQUESTED** [review]

**Review by LucianaPrat — APPROVED** [review]

**Review by NicolasGulam — APPROVED** [review]

**Pull request merged**